checks/fileparser/github_workflow.go

```diff
@@ -330,8 +330,36 @@ type JobMatcherStep struct {
 	Run string
 }
 
-// Matches returns true if the job matches the job matcher.
-func (m *JobMatcher) Matches(job *actionlint.Job) bool {
+// AnyJobsMatch returns true if any of the jobs have a match in the given workflow.
```

Comment:
`*actionlint.Workflow` Does it have to be a pointer? Harder to maintain and also moves to heap which mean GC pressure.

Reply:
My impression is that it's better for performance to pass struct parameters as pointers because the entire struct would have to be copied otherwise if it were not a pointer.

All the places that call this function have a `*actionlint.Workflow` available, so it would be a fairly large refactor to make it not a pointer. If you feel strongly these should all not use pointers, then I think that change should be made in a separate PR.

Reply:
I have seen a bunch of linters complain when we pass a large struct by value instead of pointer. I think passing by pointer here is fine.

Reply:
Yep, passing by pointer is fine here.